packages/sandpack-core/src/resolver/resolver.test.ts

```diff
@@ -503,6 +503,61 @@ describe('resolve', () => {
         '/node_modules/package-exports/src/utils/test/index.js'
       );
     });
+
+    it('should ignore the closest package.json and aliases in it when exports are available in the root', () => {
+      const resolved = resolveSync('@emotion/react/jsx-runtime', {
+        filename: '/foo.js',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      expect(resolved).toBe(
+        '/node_modules/@emotion/react/jsx-runtime/dist/emotion-react-jsx-runtime.cjs.js'
+      );
+    });
+
+    it('should always use exports from the root of the package and not from the closest package.json', () => {
+      const resolved = resolveSync('exports-from-root/nested', {
+        filename: '/foo.js',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      // this should specifically not resolve to `/node_modules/exports-from-root/nested/file.js`
+      // package.json#exports should only be used from the root of the package
+      expect(resolved).toBe('/node_modules/exports-from-root/file.js');
+    });
+
+    it('should handle conditional root exports', () => {
+      const resolved = resolveSync('its-fine', {
+        filename: '/foo.js',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      expect(resolved).toBe('/node_modules/its-fine/out/index.cjs');
+    });
+
+    it('resolve fflate correctly', () => {
```

Comment:
is this testing if a package can use a "self request"? perhaps you didn't want to use an internal `filename` here?

Reply:
Not sure I follow what you mean with internal filename, but this test has `/node_modules/its-fine/out/index.cjs` as filename intentionally as that's the real use-case that failed here

Reply:
> Not sure I follow what you mean with internal filename, but this test has /node_modules/its-fine/out/index.cjs as filename intentionally as that's the real use-case that failed here

This is kinda what I meant by the internal request - although I would call this one specifically a "self request".

1. external request: from `./src/index.js` request `fflate`
2. internal request: from `./node_modules/fflate/dist/index.js` request `./node_modules/fflate/dist/other.js`
3. self request: `./node_modules/fflate/dist/index.js` request `flatte`

Those are not official names though :P

Reply:
"resolve fflate correctly" sounds like it just tries to request `flatte` from the consuming project - but it looks to me like maybe this is testing the whole "self request" thing so maybe the test title could be more specific about that

Reply:
It's from its-fine to fflate so it is not requesting itself but a different module, but yes maybe the naming could be better, naming is hard

Reply:
ah, i totally missed that it's requesting from one package to another - makes sense. Then I guess that the self-request thing could be added as an extra test case.